hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/Text.java

```diff
@@ -301,9 +305,18 @@ public void clear() {
    */
   private boolean ensureCapacity(final int capacity) {
     if (bytes.length < capacity) {
+      // use long to allow overflow
```

Comment:
Optionally, we can also check here if capacity > MAX_ARRAY_SIZE and fail gracefully with a good error message. But it's not strictly necessary, System.arrayCopy() would fail later.